crates/outbound-pg/src/lib.rs

```diff
@@ -41,23 +45,24 @@ impl outbound_pg::OutboundPg for OutboundPg {
         statement: &str,
         params: Vec<ParameterValue<'_>>,
     ) -> Result<u64, PgError> {
-        // TODO: consider using async tokio-postgres crate
-        tokio::task::block_in_place(|| {
-            let mut client = Client::connect(address, NoTls)
-                .map_err(|e| PgError::ConnectionFailed(format!("{:?}", e)))?;
-
-            let params: Vec<&(dyn ToSql + Sync)> = params
-                .iter()
-                .map(to_sql_parameter)
-                .collect::<anyhow::Result<Vec<_>>>()
-                .map_err(|e| PgError::QueryFailed(format!("{:?}", e)))?;
-
-            let nrow = client
-                .execute(statement, params.as_slice())
-                .map_err(|e| PgError::ValueConversionFailed(format!("{:?}", e)))?;
-
-            Ok(nrow)
-        })
+        let (client, connection) = tokio_postgres::connect(address, NoTls)
+            .await
+            .map_err(|e| PgError::ConnectionFailed(format!("{:?}", e)))?;
+
+        spawn(connection);
+
+        let params: Vec<&(dyn ToSql + Sync)> = params
+            .iter()
+            .map(to_sql_parameter)
+            .collect::<anyhow::Result<Vec<_>>>()
+            .map_err(|e| PgError::QueryFailed(format!("{:?}", e)))?;
+
+        let nrow = client
+            .execute(statement, params.as_slice())
+            .await
+            .map_err(|e| PgError::ValueConversionFailed(format!("{:?}", e)))?;
```

Comment:
This might be an error in the original code - this seems more like it should be `QueryFailed` than `ValueConversionFailed`.

Reply:
Yeah, in fact it looks like this error is swapped with the one right above at line 58...

Reply:
Swapped them: https://github.com/fermyon/spin/pull/779/commits/bdbb3eed6d6d882d2a7e8585398dc625e1cbafb7

Reply:
Yikes, good catch @lann, and sorry folks, this has been wrong since the year dot.  Thanks for fixing @dicej